components/hubspot/sources/deal-updated/deal-updated.js

```diff
@@ -1,57 +1,49 @@
-const hubspot = require("../../hubspot.app.js");
+const common = require("../common.js");
 
 module.exports = {
+  ...common,
   key: "hubspot-deal-updated",
   name: "Deal Updated",
   description: "Emits an event each time a deal is updated.",
-  version: "0.0.1",
-  props: {
-    hubspot,
-    db: "$.service.db",
-    timer: {
-      type: "$.interface.timer",
-      default: {
-        intervalSeconds: 60 * 15,
-      },
+  version: "0.0.2",
+  methods: {
+    ...common.methods,
+    generateMeta(deal) {
+      const { id, properties, updatedAt } = deal;
+      const ts = Date.parse(updatedAt);
+      return {
+        id: `${id}${ts}`,
+        summary: properties.dealname,
+        ts,
+      };
+    },
+    emitEvent(deal) {
+      const meta = this.generateMeta(deal);
+      this.$emit(deal, meta);
+    },
+    isRelevant(deal, updatedAfter) {
+      return Date.parse(deal.updatedAt) > updatedAfter;
     },
   },
   async run(event) {
-    const lastRun = this.db.get("updatedAfter") || this.hubspot.monthAgo();
-    const updatedAfter = new Date(lastRun);
+    const updatedAfter =
+      this.db.get("updatedAfter") || Date.parse(this.hubspot.monthAgo());
     const data = {
       limit: 100,
       sorts: [
         {
-          propertyName: "hs_lastmodifieddate",
+          propertyName: "lastmodifieddate",
           direction: "DESCENDING",
         },
```

Comment:
Based on [the docs](https://developers.hubspot.com/docs/api/crm/deals#endpoint?spec=POST-/crm/v3/objects/deals/search), the `sorts` field is an array of strings, and the name of the "last modified date" of a deal is `hs_lastmodifieddate`.

Can you point to the documentation that made you change

Reply:
Fixed